v2/yarpcrouter/router.go

```diff
@@ -58,19 +59,78 @@ type MapRouter struct {
 func NewMapRouter(defaultService string) MapRouter {
 	return MapRouter{
 		defaultService:            defaultService,
-		serviceProcedureEncodings: make(map[serviceProcedureEncoding]yarpc.Procedure),
+		serviceProcedureEncodings: make(map[serviceProcedureEncoding]yarpc.TransportProcedure),
 		serviceNames:              map[string]struct{}{defaultService: {}},
 	}
 }
 
+// MapToTransportProcedures converts encoding-level procedures to transport-level procedures
+func MapToTransportProcedures(encodingProcedures []yarpc.EncodingProcedure) []yarpc.TransportProcedure {
+	transportProcedures := []yarpc.TransportProcedure{}
+	for _, p := range encodingProcedures {
+		var transportHandlerSpec yarpc.TransportHandlerSpec
+		switch p.HandlerSpec.Type() {
+		case yarpc.Unary:
+			transportHandlerSpec = yarpc.NewUnaryTransportHandlerSpec(unaryTransportHandler{p})
+		default:
+			panic(p.HandlerSpec.Type())
```

Comment:
Why `panic`? Should we instead return an error with a describing error message?
```
return nil, fmt.Errorf("unknown handler spec type: %v", p.HandlerSpec.Type())
```

Reply:
@kriskowal what do you think? I can't think of any reason why returning an error wouldn't be okay.

Reply:
I’m fine with whatever you all agree upon. Here’s my framework for coming to a principled answer:

We do try to avoid panics. The rule does have…exceptions. For example, clientProvider.Client(name) panics if the named client has no configuration. In that case, we would prefer to panic because it occurs during startup and indicates that the service is misconfigured, so it could not continue. Specifically, it panics *before* any lifecycles can be started. If we panic after lifecycles start, we would prefer to propagate an error and oblige Fx (for example) to stop all started components. Honestly, the exception is tenuous. Folks have asked for a MaybeClient method that does not panic, so they can safely call it at runtime.

Reply:
Currently, `clientProvider.Client(name)` returns a `yarpc.Client` and an ok `bool` so it _does not_ panic. I intentionally changed that since `Fx` would surface that error to users regardless on start up -- Frankly, I'm not a fan of panics, but I see your point here, let's talk in-person @kriskowal re: `ClientConfig`.

That being said, I would prefer to error if possible here.